Mathlib/Data/List/Rdrop.lean

```diff
@@ -0,0 +1,261 @@
+  intro h
+  have := dropWhile_nthLe_zero_not p l h
+  simp [nthLe_cons] at this
+  exact this
+
+#align list.drop_while_idempotent List.dropWhile_idempotent
+
+theorem rdrop_while_idempotent : rdropWhile p (rdropWhile p l) = rdropWhile p l :=
+  rdropWhile_eq_self_iff.mpr (rdropWhile_last_not _ _)
+#align list.rdrop_while_idempotent List.rdrop_while_idempotent
+
+/-- Take elements from the tail end of a list that satisfy `p : α → Prop`.
+Implemented naively via `List.reverse` -/
+def rtakeWhile : List α :=
+  reverse (l.reverse.takeWhile p)
+#align list.rtake_while List.rtakeWhile
+
+@[simp]
+theorem rtakeWhile_nil : rtakeWhile p ([] : List α) = [] := by simp [rtakeWhile, takeWhile]
+#align list.rtake_while_nil List.rtakeWhile_nil
+
+theorem rtakeWhile_concat (x : α) :
+    rtakeWhile p (l ++ [x]) = if p x then rtakeWhile p l ++ [x] else [] := by
+  simp only [rtakeWhile, takeWhile, reverse_append, reverse_singleton, singleton_append]
+  split_ifs with h <;> simp [h]
+#align list.rtake_while_concat List.rtakeWhile_concat
+
+@[simp]
+theorem rtakeWhile_concat_pos (x : α) (h : p x) :
+    rtakeWhile p (l ++ [x]) = rtakeWhile p l ++ [x] := by rw [rtakeWhile_concat, if_pos h]
+#align list.rtake_while_concat_pos List.rtakeWhile_concat_pos
+
+@[simp]
+theorem rtakeWhile_concat_neg (x : α) (h : ¬p x) : rtakeWhile p (l ++ [x]) = [] := by
+  rw [rtakeWhile_concat, if_neg h]
+#align list.rtake_while_concat_neg List.rtakeWhile_concat_neg
+
+theorem rtakeWhile_suffix : l.rtakeWhile p <:+ l := by
+  rw [← reverse_prefix, rtakeWhile, reverse_reverse]
+  exact takeWhile_prefix _
+#align list.rtake_while_suffix List.rtakeWhile_suffix
+
+variable {p} {l}
+
+@[simp]
+theorem rtakeWhile_eq_self_iff : rtakeWhile p l = l ↔ ∀ x ∈ l, p x := by
+  simp [rtakeWhile, reverse_eq_iff]
+#align list.rtake_while_eq_self_iff List.rtakeWhile_eq_self_iff
+
+-- Porting note: This needed a lot of rewriting to `decide` the propositions
+@[simp]
+theorem rtakeWhile_eq_nil_iff : rtakeWhile p l = [] ↔ ∀ hl : l ≠ [], ¬p (l.getLast hl) := by
+  induction' l using List.reverseRecOn with l a
+  · simp only [rtakeWhile, takeWhile, reverse_nil, true_iff]
+    intro f; contradiction
+  · simp only [rtakeWhile, reverse_append, takeWhile, reverse_eq_nil, getLast_append, ne_eq,
+  append_eq_nil, and_false, forall_true_left]
+    refine' ⟨fun h => _ , fun h => _⟩
+    · intro pa; simp only [pa, decide_True] at h
+    · simp only [h, decide_False]
+#align list.rtake_while_eq_nil_iff List.rtakeWhile_eq_nil_iff
+
+theorem mem_rtakeWhile_imp {x : α} (hx : x ∈ rtakeWhile p l) : p x := by
+  rw [rtakeWhile, mem_reverse] at hx
+  have := mem_takeWhile_imp hx
+  rw [decide_eq_true_eq] at this
+  exact this
+
+#align list.mem_rtake_while_imp List.mem_rtakeWhile_imp
+
+variable (p) (l)
+
+theorem rtakeWhile_idempotent : rtakeWhile p (rtakeWhile p l) = rtakeWhile p l :=
+  rtakeWhile_eq_self_iff.mpr fun _ => mem_rtakeWhile_imp
+#align list.rtake_while_idempotent List.rtakeWhile_idempotent
```

Comment:
I think we want to add back 
```end List``` to end the namespace

Reply:
It at least doesn't throw an error :) I have seen both variants used, but I also don't see an advantage to removing it. I'll readd it for readability.